packages/server/src/integrations/rest.ts

```diff
@@ -129,42 +130,40 @@ class RestIntegration implements IntegrationBase {
   }
 
   async parseResponse(response: any, pagination: PaginationConfig | null) {
-    let data, raw, headers
+    let data, raw, headers, filename
+
     const contentType = response.headers.get("content-type") || ""
+    const contentDisposition = response.headers.get("content-disposition") || ""
+    const matches =
+      /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/.exec(contentDisposition) || []
+    filename = matches[1]?.replace(/['"]/g, "") || ""
```

Comment:
Love a good regex.

This one doesn't appear to perfectly follow the standard for the filename parameter for `Content-Disposition`, though, sadly. Specifically this bit (from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition#filename_2):

![CleanShot 2024-03-14 at 14 14 11@2x](https://github.com/Budibase/budibase/assets/338027/6864eae5-efca-41af-a9a2-38056db305ba)

The regex takes the first thing that matches, which may or may not be the `filename*` variant.

https://stackoverflow.com/a/67994693/205770 seems to offer a potentially more robust solution, alternately https://www.npmjs.com/package/content-disposition.

We should also be wary of path-traversal attacks. In its current state, the code will parse `filename="../../../../passwd"`. I'm not sure how bad that is for MinIO but I can't imagine any legitimate reason to attempt to use `../` in filenames here, so I'd err on the side of blocking them.

Reply:
Used the content-disposition library which internally handles the `filename` / `filename*` check.